src/main.cpp

```diff
@@ -440,7 +440,166 @@ void affect_nearby_friends_with_alarm(struct Thing *traptng)
 
 long apply_wallhug_force_to_boulder(struct Thing *thing)
 {
-  return _DK_apply_wallhug_force_to_boulder(thing);
+  // return _DK_apply_wallhug_force_to_boulder(thing);
+  unsigned short angle;
+  long collide;
+  unsigned short new_angle;
+  struct Coord3d pos2;
+  struct Coord3d pos;
+  struct ShotStats *shotst = &shot_stats[thing->model];
+  unsigned short speed = (unsigned short)shotst->speed;
+  pos.x.val = thing->mappos.x.val + ((speed * lbSinTable[thing->move_angle_xy]) >> 16);
+  int n = speed * lbCosTable[thing->move_angle_xy];
+  pos.y.val = thing->mappos.y.val + (-(n >> 8) >> 8);
+  pos.z.val = thing->mappos.z.val;
+  game.action_rand_seed = ACTION_RANDOM(ULONG_MAX);
+  if ( !(game.action_rand_seed & 7) && (!thing->velocity.z.val ) )
```

Comment:
`!(game.action_rand_seed & 7) ...` should be written as `if (ACTION_RANDOM(8) == 0) && ..`